submissions/factorio_draftsman/main.md

```diff
@@ -0,0 +1,200 @@
+
+Draftsman also has support for custom `"EntityLike"` objects, most notably `Group` objects, that allow you to create custom constructs that can be inserted into blueprints for aid in clarity and compartmentalization. Again, *MAYBE* I got a little carried away.
+
+In an effort to keep this document brief, I'm not going to go into the nitty-gritty about the module or how exactly it works; I've spent a lot of time writing documentation for that purpose instead. Rather, I'm going to show off a couple of things that I've made with it so far, as well as potential things that could be made in the future, to try to illustrate exactly why I spent all this time in the first place. 
+
+## Automatic Item Stack Sizes
+
+Oftentimes for you want to figure out how much storage you need for an amount of an item. However, storage in Factorio is based on slots, not amounts, so the amount of storage is actually dependent on not only the amount you're trying to store but also the stack size of that item. [You can design a circuit contraption](https://factorioprints.com/view/-LwH68dlvOEIKWkU5JqR) to determine this amount of slots that you need, but you need a big cell of combinators documenting the item sizes. This is *boring* to make, and easily broken if a new item is added or the stack size changes. Clearly, for such a simple and repetitive task, a script is better suited:
+
+```python
+# Create a N x 5 grid of connected constant combinators, with every item and their stack size
+
+from draftsman.blueprintable import Blueprint
+from draftsman.constants import Direction
+from draftsman.data import items
+from draftsman.entity import ConstantCombinator
+
+
+def main():
+    blueprint = Blueprint()
+
+    count = 0  # Total number of signals
+    index = 0  # Signal index in the current combinator
+    i = 0  # How many combinators we've added
+    x = 0
+    y = 0
+    combinator = ConstantCombinator(direction=Direction.SOUTH)
+
+    # Iterate over every item in order:
+    for item in items.raw:
+        # Ignore hidden items/entities
+        if "flags" in items.raw[item]:
+            if "hidden" in items.raw[item]["flags"]:
+                continue
+        # Keep track of how many signals we've gone through
+        count += 1
+        # Write the stack size signal
+        stack_size = items.raw[item]["stack_size"]
+        combinator.set_signal(index, item, stack_size)
+        index += 1
+        # Once we exceed the current combinator, place it and reset
+        if index == 20:
+            # Add the combinator to the blueprint
+            combinator.id = "{}_{}".format(x, y)
+            blueprint.entities.append(combinator)
+            # Reset the combinator
+            i += 1
+            y = i % 5
```

Comment:
Shouldn't it be `y = int(i / 5)` and `x = i % 5`? I haven't played with combinators enough to know whether there's any significance to it being `5` either, but maybe turn it into a constant that is imported? Or maybe one constant for width and another for height?

Reply:
Added as global constant `COMBINATOR_HEIGHT`.